src/main/java/kea/alog/project/domain/topic/repository/TopicRepository.java

```diff
@@ -16,4 +17,7 @@ Page<Topic> findByNameContainingOrDescriptionContainingAndProjectPk(String name,
         Long projectPk,
         Pageable pageable);
 
+    Topic findByPk(Long pk);
+
+    Optional<Topic> findByPkAndProjectPk(Long pk, Long projectPk);
```

Comment:
There is no projectPk in topic entity. It should be (Long pk, Project project)

Reply:
Property expressions can refer to a direct property of the managed entity. Topic Entity has project, so it can refer to `project.pk`!

https://docs.spring.io/spring-data/jpa/docs/current/reference/html/
I recommend you to refer to the relevant data in ***4.4.3. Property Expressions*** above link😺